packages/venia-concept/venia-upward.yml

```diff
@@ -68,97 +122,187 @@ isDevelopment:
+    # a generic shell as the template file.
     - matches: urlResolver.type
       pattern: '.'
       use:
         inline:
           model: unknownPageType
           template: './templates/generic-shell.mst'
+    # This entry makes '/search.html' a valid resource path.
+    # It uses the top-level 'genericObject' object as its resolve data and uses
+    # the generic shell as its template file.
     - matches: request.url.pathname
       pattern: '/search.html'
       use:
         inline:
           model: genericObject
           template: './templates/generic-shell.mst'
   default:
+    # If none of the resolver conditions pass, no 'model' property is set and
+    # this resource object sets the template property to the notfound-shell
+    # template file.
     inline:
       template: './templates/notfound-shell.mst'
 
+# The assetManifest object is a FileResolver that returns JSON data from a
+# file generated by Webpack. This file contains a list of all the "bundle"
+# and "chunk" filenames generated by the last Webpack build. Using this data,
+# the templates can render dynamic script tags based on the filenames of the
+# most recent build assets.
+#
+# NOTE: This object uses the shorthand syntax for a FileResolver to help with 
+# readability and as a convenience.
 assetManifest:
   when:
     - matches: env.NODE_ENV
       pattern: 'development'
       use: ./asset-manifest.json
   default: ./dist/asset-manifest.json
 
+# The unknownPageType object contains a single 'title' property that is used
+# in the template file as the page title.
+# The top-level 'resource' object uses this object as the model data for page
+# types it does not support.
 unknownPageType:
   inline:
     title: 'Unknown page type'
 
+# The urlResolver object is an object containing only the relevant
+# data from the top-level 'urlResolverResult' object, which is a
+# ServiceResolver.
+#
+# NOTE: Extracting smaller objects from large or deeply nesting data objects
+# is a pattern used to enhance readability and avoid long "dot lookup" strings
 urlResolver: urlResolverResult.data.urlResolver
 
+# The urlResolverResult object is a ServiceResolver that returns data from a
+# GraphQL backing service, whose URL is resolved in the top-level 'magentoGQL'
+# object.
+#
+# This object retrieves information from Magento about the request path,
+# so the PWA can respect redirects and routes controlled by the admin.
+#
+# NOTE: The UPWARD server is able to interpret this entry as a ServiceResolver
+# because of the presence of the 'query' property, which itself uses the
+# shorthand syntax for a FileResolver.
 urlResolverResult:
   url: magentoGQL
   query: './src/queries/urlResolver.graphql'
   variables:
     inline:
       urlKey: request.url.pathname
 
+# The magentoGQL object uses a TemplateResolver to generate the URL for the
+# GraphQL endpoint using the MAGENTO_BACKEND_URL environment variable.
```

Comment:
We replaced template resolution of URLs with the new URL Resolver. Let's merge in mainline and update these sections.

Reply:
Merged and updated